roxie/udplib/udptrr.cpp

```diff
@@ -312,127 +300,79 @@ class CReceiveManager : implements IReceiveManager, public CInterface
-                enqueueRequest(requester);
-            if (pendingRequests)
-                return sendNextOk();
+            if (requester->state == flowType::ok_to_send)
+            {
+               pendingPermits.remove(requester);
+               requester->state = flowType::send_completed;
+            }
             else
-                return 5000;
+            {
+                StringBuffer s;
+                DBGLOG("Received unexpected completed message (state is %s) from %s", flowType::name(requester->state), requester->dest.getIpText(s).str());
+            }
         }
 
-
-        unsigned sendNextOk()
+        unsigned slotsAvailable()
         {
-            assert(pendingRequests != nullptr);
-            if (udpSnifferEnabled)
-            {
-                //find first non-busy sender, and move it to front of sendersTable request chain
-                int retry = udpRetryBusySenders;
-                UdpSenderEntry *finger = pendingRequests;
-                UdpSenderEntry *prev = nullptr;
-                for (;;)
-                {
-                    if (finger->is_busy())
-                    {
-                        prev = finger;
-                        finger = finger->nextSender;
-                        if (finger==nullptr)
-                        {
-                            if (retry--)
-                            {
-                                if (udpTraceLevel > 4)
-                                    DBGLOG("UdpReceive: All senders busy");
-                                MilliSleep(1);
-                                finger = pendingRequests;
-                                prev = nullptr;
-                            }
-                            else
-                                break; // give up and use first anyway
-                        }
-                    }
-                    else
-                    {
-                        if (finger != pendingRequests)
-                        {
-                            if (finger == lastPending)
-                                lastPending = prev;
-                            assert(prev != nullptr);
-                            prev->nextSender = finger->nextSender;
-                            finger->nextSender = pendingRequests;
-                            pendingRequests = finger;
-                        }
-                        break;
-                    }
-                }
-            }
-            UdpSenderEntry *nextSender = pendingRequests;
-            // remove from front of queue
-            if (pendingRequests==lastPending)
-                lastPending = nullptr;
-            pendingRequests = nextSender->nextSender;
-            nextSender->nextSender = nullptr;
-            return okToSend(nextSender);
+            unsigned slots = parent.input_queue->available();
+            if (slots>maxSlotsPerSender)
+                slots = maxSlotsPerSender;
+//          if (slots<minSlotsPerSender)
```

Comment:
Idea is that we should wait until we can ask for >1, to reduce handshake overhead. But not yet implemented this

Reply:
remove, or make into a more explicit comment?

Reply:
Actually, function is not called any more, so remove...

Reply:
Removed